### ISSUE TITLE
darktable 3.0.2. crashing while using the color zones module

### ISSUE THREAD

**Issue by RicoRichardson** [darktable-org/darktable#5163]
Hey guys,

Darktable has crashed on me while trying to use the color zones module.
It happened while adjusting the lightness of the pixels.
I've attached the backtrace file. Hopefully that helps.
Personally, I don't know anything about code so I hope this will be enough information.

Greets,

Rico
[darktable_bt_MXZ8K0.txt](https://github.com/darktable-org/darktable/files/4669448/darktable_bt_MXZ8K0.txt)

**Comment by johnny-bit** [comment]
Hi Rico, can you attach XMP file produced by crash?

Also couple of things:
OpenCL: Was it on or off?
What OS do you use?
What graphics card and driver version?

Copuple other things:
1. Can you reliably replicate the issue on another file?
2. Does issue happen on JPEGs too?
3. can you replicate the issue fro scratch (clear history)?
4. Can you temporally remove config+library (move them to safe place) and start fresh?

**Comment by RicoRichardson** [comment]
Hello!

OpenCL: No clue, but I guess so?
OS: Windows 10
Graphics card: Nvidia 1060 6GB (Max-Q design) driver version: 441.41

System I'm using:
Dell 7577
i7700-HQ (2.8Ghz)
32GB RAM
3840*2160 60Hz display

When uploading the XMP file I get a message that the file type isn't supported. So I've put it in a .txt for you. Hope that's good.
[IMG_0786.txt](https://github.com/darktable-org/darktable/files/4670314/IMG_0786.txt)


Other things:
1. I'll have to try. What I can replicate is the crash darktable has when opening up a RAW file. Doens't matter if it's .NEF, .CR2, etc. The only way I can open up files now is by going into darktable first (but this is a different issue, and already known right?)
2. Haven't had this issue on .jpg yet
3. You mean with a clean file that doesn't have an .xmp yet?
4. You mean a reinstall of darktable? I just did and that's when it crashed.

**Comment by johnny-bit** [comment]
Wow, this is weird. that XMP file is "empty" meaning - it has no adjustments what so ever.

So now is seems that best case of action would be to backup directory `C:\Users\[username]\AppData\Local\darktable` (substitute `[username]` with your username) and remove it. You can simply rename it to eg `darktable_backup` or something and then start darktable. It should start with clean config and clean library (no images imported, no styles etc). If then you'll try to open up RAW and there won't be a problem then most likelly you have some auto-applied preset that breaks stuff.

**Comment by RicoRichardson** [comment]
Yeah that's the thing.
When darktable gives me a crash like that, it remembers NON of the adjustments I've made to the image. I have to redo everything. Before 3.0.2 this would happen when using masks every now and then. This time I wasn't using masks. So it's odd.

I've just tried to replicate the error again by using the tone equalizer first, then the color zones (first hue, then lightness) but it didn't crash on me. 


What can be the auto-applied preset? Since I don't really use presets at all. The only thing I ALWAYS do is go to the orientation step, and compress the history stack to start from scratch.
